lib/ansible/galaxy/collection.py

```diff
@@ -578,6 +578,10 @@ def verify_collections(collections, search_paths, apis, validate_certs, ignore_e
                             break
                     if local_collection is None:
                         raise AnsibleError(message='Collection %s is not installed in any of the collection paths.' % collection_name)
+                    if local_collection.latest_version == '*':
+                        msg = 'Collection %s does not appear to have the MANIFEST.json. ' % collection_name
+                        msg += 'A MANIFEST.json is expected if the collection has been built and installed via ansible-galaxy.'
+                        raise AnsibleError(message=msg)
```

Comment:
My only minor problem is that now we show both a warning and an error that essentially say the same thing.
```
[WARNING]: Collection at '/Users/sdoran/.ansible/collections/ansible_collections/openstack/cloud' does not have a MANIFEST.json file,
cannot detect version.
ERROR! Collection openstack.cloud does not appear to have the MANIFEST.json. A MANIFEST.json is expected if the collection has been built and installed via ansible-galaxy.
```

Since these are coming from two different places, there isn't currently a clear way to suppress the warning because we may later raise an error. That's my only minor issue with this change.

Reply:
I wondered about that too. I added the exception to halt execution (unless ignore errors is used)... but looking at this again, I could instead check os.path.isfile for the MANIFEST.json before calling from_path() and then we'd get rid of the duplicate.

Reply:
That should be fixed in 470dfbe776cb51be353154f7697555cebc995707.

Reply:
Awesome. This is much better now.